src/main/java/org/wildfly/security/x500/X500PrincipalUtil.java

```diff
@@ -41,15 +42,22 @@
 public final class X500PrincipalUtil {
 
     private static final String[] NO_STRINGS = new String[0];
-    private static final boolean HAS_X500_NAME;
+    private static final Class<?> X500_NAME_CLASS;
+    private static final MethodHandle AS_X500_PRINCIPAL_HANLDE;
 
     static {
-        boolean hasX500Name = false;
+        Class<?> x500Name = null;
+        MethodHandle asX500PrincipalHandle = null;
         try {
-            Class.forName("sun.security.x509.X500Name", true, X500PrincipalUtil.class.getClassLoader());
-            hasX500Name = true;
-        } catch (Throwable t) {}
-        HAS_X500_NAME = hasX500Name;
+            x500Name = Class.forName("sun.security.x509.X500Name", true, X500PrincipalUtil.class.getClassLoader());
+            asX500PrincipalHandle = MethodHandles.publicLookup().unreflect(x500Name.getDeclaredMethod("asX500Principal"));
+        } catch (ClassNotFoundException e) {
+            /* expected on non OpenJDK/Oracle based runtimes */
+        } catch (IllegalAccessException | NoSuchMethodException e) {
+            throw new RuntimeException(e);
```

Comment:
It's better to wrap or translate these into their Error equivalents rather than just wrapping in RuntimException.